backend/inspirehep/pidstore/minters/texkey.py

```diff
@@ -59,8 +60,24 @@ def prepare_and_mint(cls, object_uuid, data):
             minter.add_texkey(pid_value)
         # Run it without pid provided to check if texkey should be re-generated
         minter.add_texkey(None)
+        # Delete from pidstore texkeys which are not in metadata
+        minter.purge_missing()
         return minter
 
+    def purge_missing(self):
+        texkeys = (
+            PersistentIdentifier.query.filter(
+                PersistentIdentifier.object_uuid == self.object_uuid
```

Comment:
```suggestion
                PersistentIdentifier.object_uuid == self.object_uuid,
                PersistentIdentifier.object_type == "rec",
```
to benefit from the index.

Reply:
:+1: